src/uct/base/uct_iface.c

```diff
@@ -363,11 +363,13 @@ ucs_status_t uct_set_ep_failed(ucs_class_t *cls, uct_ep_h tl_ep,
 
     if (iface->err_handler) {
         return iface->err_handler(iface->err_handler_arg, tl_ep, status);
+    } else if (status == UCS_ERR_CANCELED) {
+        status = UCS_OK;
```

Comment:
why isn't err handling invoked from here?
it should only be called from flush?

Reply:
If error handler is set, it will be invoked above, otherwise the error is not handled and this causes fatal error on transport level. Since the cancelation was initiated by user, we may assume that it's OK.

Reply:
so we dont care about any error flows that happened prior the user's cancellation?
why is error handling invoked then here https://github.com/openucx/ucx/pull/3408/files#diff-73e0f9c930e2212de847a8d24c0c0e05R53 ?

Reply:
> so we dont care about any error flows that happened prior the user's cancellation?

sorry, don't understand how it's related.. prior errors will be handled prior, right?

> why is error handling invoked then here 

to cleanup SW and HW resources.